include/onnxruntime/core/framework/sparse_tensor.h

```diff
@@ -1,72 +1,427 @@
-#pragma once
-
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#pragma once
+
 #include "core/framework/data_types.h"
 #include "core/framework/tensor_shape.h"
 #include "core/framework/tensor.h"
 
 namespace onnxruntime {
 
+class IDataTransfer;
+class DataTransferManager;
+
+/**
+ * @brief This is a Sparse Format enumeration representing bitflags
+ * 
+ * 
+ */
+enum class SparseFormat : uint32_t {
+  kUndefined = 0x0U,        // For completeness
+  kCoo = 0x1U,              // 1-D or 2-D indices
+  kCsrc = 0x1U << 1,        // Both CSR(C)
+  kBlockSparse = 0x1U << 2  // as in OpenAI
+};
+
+std::ostream& operator<<(std::ostream&, SparseFormat);
+
 /**
  * @brief This class implements SparseTensor. 
- * We represent a SparseTensor as a triple <values, indices, shape>. "values" and "indices" themselves
- * are implemented as Tensors. 
- * We follow the Tensor design for memory ownership/management: a sparse-tensor does not own the "value"
- * or "indices" tensors.
+ * We represent a SparseTensor as a single contiguous buffer with Tensors that project
+ * values into that buffer with other Tensors projecting format specific indices.
+ * 
  */
 
 class SparseTensor final {
```

Comment:
Can we document the usage of this class somewhere? Will make both code reviews and using it easier.